### PR TITLE
Changed distance function last argument type

### PR DESCRIPTION
**Pull request by meiravgri**
In the current implemntation, distance functions last argument was of type const void *, which was converted to size_t in the function body
Since it is only used to pass the vector dimensions, we changed the type to size_t.

This change saves casting + derefrencing the dim parameter each time we call a function to calculate distance between two vectors.

### CODE REVIEW

File: src/VecSim/spaces/spaces.h [hunk 1]
```diff
@@ -8,7 +8,7 @@ static const Arch_Optimization arch_opt = getArchitectureOptimization();
 
 // TODO change last arg to size_t
```
Comment (GuyAv46) [inline src/VecSim/spaces/spaces.h:9]: Done :)

File: src/VecSim/spaces/IP/IP_AVX.cpp [hunk 2]
```diff
@@ -1,16 +1,14 @@
 
+#include <cstdlib>
+
 #include "IP_AVX.h"
 #include "VecSim/spaces/space_includes.h"
 #include "VecSim/spaces/IP/IP.h"
```
Comment (GuyAv46) [inline src/VecSim/spaces/IP/IP_AVX.cpp:6]: Are all of these needed?

**Review by GuyAv46 — APPROVED** [review]
Looking good!
Consider going over some of the includes and reorganize them.
also please mark this PR for benchmark and see if there is an effect on the distance functions benchmarks

File: src/VecSim/spaces/IP/IP_AVX.cpp (hunk 2 above)
Comment (meiravgri) [reply src/VecSim/spaces/IP/IP_AVX.cpp]: yes (except stdlib)

**Review by DvirDukhan — APPROVED** [review]

**Pull request merged**